### PR TITLE
Resolved cyclic dependencies between some parts of the lib

### PR DESCRIPTION
**Pull request by KOLANICH**
Resolved cyclic dependencies between some parts of the lib in order to allow dynamic linking to be used on Windows platform. Also resolved some issues related to CPack packaging.

### CODE REVIEW

File: src/io/io.cpp [hunk 1]
```diff
@@ -15,6 +15,9 @@
```
Comment (alex19EP) [inline src/io/io.cpp:15]: you need to fix scons build system if you want to move this file.

Comment (alex19EP) [reply src/io/io.cpp:15]: src/question_match/question_matcher.cpp
and this too.

Comment (KOLANICH) [reply src/io/io.cpp:15]: I know, just haven't done that yet.

Comment (KOLANICH) [reply src/io/io.cpp:15]: Fixed SCONS for linux, but haven't fixed java and Windows pipelines yet.

**Pull request closed without merging**